finality-aleph/src/party.rs

```diff
@@ -54,21 +57,34 @@ where
                 auth_keystore,
                 justification_rx,
                 metrics,
-                period,
+                session_period,
+                millisecs_per_block,
                 ..
             },
     } = aleph_params;
 
     let sessions = Arc::new(Mutex::new(HashMap::new()));
 
+    // NOTE: justifications are requested every so often
+    let justifications_cadence = min(
+        (session_period.0 as u64 * 1000u64) * millisecs_per_block.0,
+        millisecs_per_block.0,
+    ) / 1000;
```

Comment:
This is not the same as the formula mentioned in the task description, we definitely don't want this to be as frequent as block production.

Also, be careful with integer division, we don't want the delay to be 0 if the result turns out to be subsecond.

Reply:
Also isn't `millisecs_per_block.0` always smaller than `(session_period.0 as u64 * 1000u64) * millisecs_per_block.0`?
`(session_period.0 as u64 * 1000u64)` is an integer (and I don't think it makes sense when set to 0).

Reply:
Maybe instead of division we could use `JustificationsCadence::from_millis(...)`?

Reply:
good point, although technically we never assert that session is longer than time till block production :P

Reply:
The formula is changed now